test/common/upstream/hds_test.cc

```diff
@@ -95,16 +100,66 @@ class HdsTest : public testing::Test {
         envoy::type::v3::HTTP1);
     health_check->mutable_health_checks(0)->mutable_http_health_check()->set_path("/healthcheck");
 
-    auto* socket_address = health_check->add_locality_endpoints()
-                               ->add_endpoints()
-                               ->mutable_address()
-                               ->mutable_socket_address();
+    auto* locality_endpoints = health_check->add_locality_endpoints();
+    // add locality information to this endpoint set of one endpoint.
+    locality_endpoints->mutable_locality()->set_region("middle_earth");
+    locality_endpoints->mutable_locality()->set_zone("shire");
+    locality_endpoints->mutable_locality()->set_sub_zone("hobbiton");
+
+    // add one endpoint to this locality grouping.
+    auto* socket_address =
+        locality_endpoints->add_endpoints()->mutable_address()->mutable_socket_address();
     socket_address->set_address("127.0.0.0");
     socket_address->set_port_value(1234);
 
     return msg;
   }
 
+  // Creates a HealthCheckSpecifier message that contains several clusters, endpoints, localities,
+  // with only one health check type.
+  envoy::service::health::v3::HealthCheckSpecifier*
+  createComplexSpecifier(const int n_clusters, const int n_localities, const int n_endpoints) {
+    // Final specifier to return.
+    envoy::service::health::v3::HealthCheckSpecifier* msg =
+        new envoy::service::health::v3::HealthCheckSpecifier;
+
+    // set interval.
+    msg->mutable_interval()->set_seconds(1);
+
+    for (int cluster_num = 0; cluster_num < n_clusters; cluster_num++) {
+      // add a cluster with a name by iteration, with path /healthcheck
+      auto* health_check = msg->add_cluster_health_checks();
+      health_check->set_cluster_name("anna" + std::to_string(cluster_num));
+      health_check->add_health_checks()->mutable_timeout()->set_seconds(1);
+      health_check->mutable_health_checks(0)->mutable_interval()->set_seconds(1);
```

Comment:
suggest capturing `health_check->mutable_health_checks(0)` in a temp.

Reply:
Thanks, temp is now being used here and in similar places.